apis/v1beta1/shared_types.go

```diff
@@ -150,7 +172,14 @@ type CommonRouteSpec struct {
 	// rules. Cross-namespace references are only valid if they are explicitly
 	// allowed by something in the namespace they are referring to. For example,
 	// Gateway has the AllowedRoutes field, and ReferenceGrant provides a
-	// generic way to enable any other kind of cross-namespace reference.
+	// generic way to enable other kinds of cross-namespace reference.
+	//
+	// ParentRefs from a Route to a Service in the same namespace are "producer"
+	// routes, which apply default routing rules to inbound connections from
+	// any namespace to the Service. ParentRefs from a Route to a Service in a
+	// different namespace are "consumer" routes, and these routing rules are
+	// only applied to outbound connections to the Service from the same
+	// namespace as the Route.
```

Comment:
Just so it doesn't get missed, this paragraph should be updated to match the related changes you're proposing above whenever we reach consensus on those.